deker/ABC/base_array.py

```diff
@@ -459,25 +459,35 @@ def _create_from_meta(
             attrs_schema = collection.varray_schema.attributes
         else:
             attrs_schema = collection.array_schema.attributes
+
         try:
-            for attr in attrs_schema:
-                attributes = (
-                    meta["primary_attributes"] if attr.primary else meta["custom_attributes"]
-                )
+            # To ensure the order of attributes
+            primary_attributes, custom_attributes = OrderedDict(), OrderedDict()
```

Comment:
Custom attributes are not ordered. Why do we need OrderedDict here for them?

Reply:
They have same logic as primary. Our attributes is just a list where attributes are placed one by one. Some of them are primary some not. I think it's better to keep same logic for both kinds, as we might need ordered custom attributes in future and anyway we iterate over every single attribute, so why not order them as well. We don't loose anything from doing it.